src/world/Movement/MovementGenerators/IdleMovementGenerator.cpp

```diff
@@ -152,6 +172,16 @@ void DistractMovementGenerator::initialize(Unit* owner)
     init.SetFacing(_orientation);
     init.Launch();
 }
+#else
+void DistractMovementGenerator::initialize(Unit* owner)
+{
+    // Distracted creatures stand up if not standing
+    if (!owner->getStandState())
+        owner->setStandState(STANDSTATE_STAND);
+
+    owner->addUnitStateFlag(UNIT_STATE_DISTRACTED);
```

Comment:
Is this state flag needed in classic - wotlk?

Reply:
Some Lines Above its in the BaseUnitState, and there we launch a MoveSplineInit and this uses the initialized State